PLUGIN-517

ardian4 opened the pull request:
Added: Test BigQuery Sink when destination table does not exist, partition type is INTEGER, and partition field is macro
Added: Test BigQuery Sink when destination table does exist, partition type is INTEGER, and partition field is macro

Jira Ticket: https://cdap.atlassian.net/browse/PLUGIN-517
PR: https://github.com/data-integrations/google-cloud/pull/553

flakrimjusufi commented on integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java:
```diff
@@ -877,6 +879,267 @@ public void testUpdateOperationWithSchemaUpdate() throws Exception {
+  private void testInsertOperationWithIntegerPartition(Engine engine) throws Exception {
+    String testId = GoogleBigQueryUtils.getUUID();
+
+    String sourceTableName = SOURCE_TABLE_NAME_TEMPLATE + testId;
```
We can use String.format() in here.

flakrimjusufi commented on integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java:
```diff
@@ -877,6 +879,267 @@ public void testUpdateOperationWithSchemaUpdate() throws Exception {
+
+    checkMetric(tags, "user." + deploymentDetails.getSource().getName() + ".records.out", expectedCount, 10);
+    checkMetric(tags, "user." + deploymentDetails.getSink().getName() + ".records.in", expectedCount, 10);
+
```
Same in here. We can use String.format() for checkMetric.

flakrimjusufi commented on integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java:
```diff
@@ -877,6 +879,267 @@ public void testUpdateOperationWithSchemaUpdate() throws Exception {
+
+    checkMetric(tags, "user." + deploymentDetails.getSource().getName() + ".records.out", expectedCount, 10);
+    checkMetric(tags, "user." + deploymentDetails.getSink().getName() + ".records.in", expectedCount, 10);
+
```
Same in here.

flakrimjusufi reviewed (commented):
Some minor comments. 
Please check them out and we can go for the next round of review.

flakrimjusufi reviewed (commented):
Looks OK to me.

rmstar commented on integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java:
```diff
@@ -877,6 +879,268 @@ public void testUpdateOperationWithSchemaUpdate() throws Exception {
+   *  {"string_value":"string_2","int_value":2,"float_value":0.2,"boolean_value":false}
+   *  {"string_value":"string_3","int_value":3,"float_value":0.3,"boolean_value":false}
+   * Starting sink data:
+   *  {"string_value":"string_1","int_value":1,"float_value":0.1,"boolean_value":true}
```
How is Starting sink data same as Expected ending sink data?

ardian4 replied:
fixed

rmstar reviewed (commented):
Can we also add tests for time partitioning?

ardian4 commented:
> Can we also add tests for time partitioning?

Added  test when partition field is empty and in BQ table will be created as Ingestion Time Partition
Added  test when partition field is not empty and in BQ table will be created as Date Time Column Partition

rmstar reviewed (approved):
please squash commits
